front/lib/iam/session.ts

```diff
@@ -75,3 +83,44 @@ export async function getUserFromSession(
     }),
   };
 }
+
+interface WithGetServerSidePropsRequirementsOptions {
+  enableLogging?: boolean;
+  requireAuth?: boolean;
+}
+
+const defaultWithGetServerSidePropsRequirements: WithGetServerSidePropsRequirementsOptions =
+  {
+    enableLogging: true,
+    requireAuth: true,
+  };
+
+export function withGetServerSidePropsRequirements<
+  T extends { [key: string]: any } = { [key: string]: any }
+>(
+  getServerSideProps: GetServerSideProps<T>,
+  opts: WithGetServerSidePropsRequirementsOptions = defaultWithGetServerSidePropsRequirements
+): GetServerSideProps<T> {
+  return async (
+    context: GetServerSidePropsContext<ParsedUrlQuery, PreviewData>
+  ) => {
+    const { enableLogging, requireAuth } = opts;
+
+    const session = await getSession(context.req, context.res);
```

Comment:
do we need to get the session if `requireAuth` is false ?

Reply:
We could indeed remove it, in a next PR I'm providing the session down to the `getServerSideProps` function so we don't call `getSession` everywhere.

Reply:
Fixed in https://github.com/dust-tt/dust/pull/4129/commits/f2f5157ec0fb19135f8dc207eac5a8f4a5e52093.